### PR TITLE
Flash fixed

### PR DESCRIPTION
**Pull request by mosheraboh**
# What does this PR do?

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: src/transformers/models/t5/modeling_t5.py [hunk 1]
```diff
@@ -91,14 +92,17 @@ def rotate_half(x):
...
+def apply_rotary_pos_emb(x, cos, sin, seq_dimension):
     # NOTE: This could probably be moved to Triton
 
-    # Handle a possible sequence length mismatch in between q and k
-    cos = cos[:, :, : x.shape[-2], :]
-    sin = sin[:, :, : x.shape[-2], :]
+    # # Handle a possible sequence length mismatch in between q and k
+    # cos = cos[:, :, : x.shape[-2], :]
+    # sin = sin[:, :, : x.shape[-2], :]
+    res = torch.clone(x)
```
Comment (YoelShoshan) [inline src/transformers/models/t5/modeling_t5.py:101]: why the clone here?

File: src/transformers/models/t5/modeling_t5.py [hunk 2]
```diff
@@ -768,19 +768,20 @@ def forward(
...
             if mask is not None:
                 position_bias = position_bias + mask  # (batch_size, n_heads, seq_length, key_length)
 
-            if self.pruned_heads:
-                mask = torch.ones(position_bias.shape[1])
-                mask[list(self.pruned_heads)] = 0
-                position_bias_masked = position_bias[:, mask.bool()]
-            else:
-                position_bias_masked = position_bias
+        if self.pruned_heads:
```
Comment (YoelShoshan) [inline src/transformers/models/t5/modeling_t5.py:777]: Great!

File: src/transformers/models/t5/modeling_t5.py [hunk 3]
```diff
@@ -2019,7 +2020,9 @@ def forward(
         >>> print(tokenizer.decode(outputs[0], skip_special_tokens=True))
         >>> # studies have shown that owning a dog is good for you.
         ```"""
-
+        # temp fix
+        use_cache = False
```
Comment (YoelShoshan) [inline src/transformers/models/t5/modeling_t5.py:2024]: do you want it to stay like this ?

**Review by YoelShoshan — APPROVED** [review]
Looks great!
Awesome work spotting and solving those multiple tricky bugs!

It does make me think that we HAVE to add live unit tests that do regression tests in which model weights are loaded from a checkpoint, a random input tensor is fed in, and the output is compared to a saved numpy array to spot regression.

Initially maybe just for CPU, and later also for different combinations of GPU type and precision.

File: src/transformers/models/t5/modeling_t5.py (hunk 3 above)
Comment (mosheraboh) [reply src/transformers/models/t5/modeling_t5.py:2024]: Yes, until we will get back and fix this issue or if Michal says it takes too long to generate :)

File: src/transformers/models/t5/modeling_t5.py (hunk 1 above)
Comment (mosheraboh) [reply src/transformers/models/t5/modeling_t5.py:101]: It's probably not required - but since x is cached, I did want to touch it to be on the safe side.

**Pull request merged**